### PR TITLE
osdocs-478 machine support GCP

### PR DESCRIPTION
**Pull request by kalexand-rh**
https://jira.coreos.com/browse/OSDOCS-478

@enxebre, PTAL?

### CODE REVIEW

File: modules/machineset-yaml-gcp.adoc [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        value:
+          apiVersion: gcpprovider.openshift.io/v1beta1
+          canIPForward: false
+          credentialsSecret:
+            name: gcp-cloud-credentials
+          deletionProtection: false
+          disks:
+          - autoDelete: true
+            boot: true
+            image: <infrastructureID>-rhcos-image
```
Comment (kalexand-rh) [inline modules/machineset-yaml-gcp.adoc]: This image is static for your entire cluster, right?

File: modules/machineset-yaml-gcp.adoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            labels: null
+            sizeGb: 128
+            type: pd-ssd
+          kind: GCPMachineProviderSpec
+          machineType: n1-standard-4
+          metadata:
+            creationTimestamp: null
+          networkInterfaces:
+          - network: <infrastructureID>-network
+            subnetwork: <infrastructureID>-worker-subnet
```
Comment (kalexand-rh) [inline modules/machineset-yaml-gcp.adoc]: If you're making a new machineset, what do you specify for the subnet?

File: modules/machineset-yaml-gcp.adoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          projectID: openshift-gce-devel
+          region: us-central1
+          serviceAccounts:
+          - email: <infrastructureID>-w@openshift-gce-devel.iam.gserviceaccount.com
```
Comment (kalexand-rh) [inline modules/machineset-yaml-gcp.adoc]: How is this value generated? Do you need to generate a different value for a new machineset up front?

File: modules/machineset-yaml-gcp.adoc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            scopes:
+            - https://www.googleapis.com/auth/cloud-platform
+          tags:
+          - <infrastructureID>-<role> <3>
+          userDataSecret:
+            name: worker-user-data
```
Comment (kalexand-rh) [inline modules/machineset-yaml-gcp.adoc]: If you're making a new machineset, what do you specify for the userDataSecret? Do you omit this parameter?

**Review by kalexand-rh — COMMENTED** [review]
@enxebre, I have some questions about these parameters.
Are there any others that need to change when you modify the machineset?

**Comment by kalexand-rh** [comment]
@openshift/team-documentation PTAL

File: modules/machineset-yaml-gcp.adoc (hunk 1 above)
Comment (enxebre) [reply modules/machineset-yaml-gcp.adoc]: not necessarily, but it will most of the times

File: modules/machineset-yaml-gcp.adoc (hunk 2 above)
Comment (enxebre) [reply modules/machineset-yaml-gcp.adoc]: you can re-use the values in the original machineSet

File: modules/machineset-yaml-gcp.adoc (hunk 4 above)
Comment (enxebre) [reply modules/machineset-yaml-gcp.adoc]: you can re-use `worker-user-data`

File: modules/machineset-yaml-gcp.adoc (hunk 1 above)
Comment (kalexand-rh) [reply modules/machineset-yaml-gcp.adoc]: When would it be different?

File: modules/machineset-yaml-aws.adoc [hunk 5]
```diff
@@ -63,20 +64,26 @@ spec:
...
+package installed, you can obtain the infrastructure ID by running the following
+command:
++
+----
+$ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
+----
+<2> Specify the infrastructure ID and node label.
 <3> Specify the node label to add.
-<4> Specify a valid {op-system-first} AMI for your Amazon Web Services (AWS) zone for your {product-title} nodes.
+<4> Specify a valid {op-system-first} AMI for your AWS zone for your {product-title} nodes.
```
Comment (bobfuru) [inline modules/machineset-yaml-aws.adoc]: Do we spell out "AMI" on first instance?

**Review by bobfuru — COMMENTED** [review]
Couple of small suggestions; otherwise, LGTM!

File: machine_management/creating_machinesets/creating-machineset-aws.adoc [hunk 6]
```diff
@@ -0,0 +1,17 @@
+[id="creating-machineset-aws"]
+= Creating a MachineSet in AWS
+include::modules/common-attributes.adoc[]
+:context: creating-machineset-aws
+
+You can create a different MachineSet to serve a specific purpose in your
+{product-title} cluster on Amazon Web Services (AWS). For example, you might
+create infrastructure MachineSets and related Machines so that you can move
+supporting workloads to it.
```
[the remark this answers is not in the document]
Comment (kalexand-rh) [reply machine_management/creating_machinesets/creating-machineset-aws.adoc]: "It" is the Machines, not the cluster. I'll make that clearer.

File: modules/machineset-yaml-gcp.adoc [hunk 7]
```diff
@@ -0,0 +1,85 @@
+/ Module included in the following assemblies:
```
Comment (sunzhaohua2) [inline modules/machineset-yaml-gcp.adoc]: This line can be deleted.

File: modules/machineset-yaml-gcp.adoc [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+//
+// * machine_management/creating-infrastructure-machinesets.adoc
+// * machine_management/creating-machineset-gcp.adoc
+
+[id="machineset-yaml-gcp_{context}"]
+=  Sample YAML for a MachineSet Custom Resource on GCP
+
+This sample YAML defines a MachineSet that runs in Google Cloud Platform (GCP_
```
Comment (sunzhaohua2) [inline modules/machineset-yaml-gcp.adoc]: shoud be (GCP) not (GCP_

File: modules/machineset-yaml-gcp.adoc [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+the cluster, and `<role>` is the node label to add.
+
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructureID> <1>
+  name: <infrastructureID>-w-a <2>
```
Comment (sunzhaohua2) [inline modules/machineset-yaml-gcp.adoc]: name: <infrastructureID>-w-a <1>

File: modules/machineset-yaml-gcp.adoc [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+  namespace: openshift-machine-api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <infrastructureID> <1>
+      machine.openshift.io/cluster-api-machineset: <infrastructureID>-w-a <2>
```
Comment (sunzhaohua2) [inline modules/machineset-yaml-gcp.adoc]: machine.openshift.io/cluster-api-machineset: <infrastructureID>-w-a <1>

File: modules/machineset-yaml-gcp.adoc [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        machine.openshift.io/cluster-api-cluster: <infrastructureID> <1>
+        machine.openshift.io/cluster-api-machine-role: worker
+        machine.openshift.io/cluster-api-machine-type: worker
+        machine.openshift.io/cluster-api-machineset: <infrastructureID>-w-a <2>
```
Comment (sunzhaohua2) [inline modules/machineset-yaml-gcp.adoc]: machine.openshift.io/cluster-api-machineset: <infrastructureID>-w-a <1>

File: modules/machineset-yaml-gcp.adoc [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+            image: <infrastructureID>-rhcos-image <1>
+            labels: null
+            sizeGb: 128
+            type: pd-ssd
+          kind: GCPMachineProviderSpec
+          machineType: n1-standard-4
+          metadata:
+            creationTimestamp: null
+          networkInterfaces:
+          - network: <infrastructureID>-network
```
Comment (sunzhaohua2) [inline modules/machineset-yaml-gcp.adoc]: - network: <infrastructureID>-network <1>

File: modules/machineset-yaml-gcp.adoc [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+            subnetwork: <infrastructureID>-worker-subnet <1>
```
Comment (sunzhaohua2) [inline modules/machineset-yaml-gcp.adoc]: subnetwork: <infrastructureID>-<role>-subnet <2>

File: modules/machineset-yaml-gcp.adoc [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+          projectID: openshift-gce-devel
```
Comment (sunzhaohua2) [inline modules/machineset-yaml-gcp.adoc]: The projectID to use depends on your own project.

File: modules/machineset-yaml-gcp.adoc (hunk 9 above)
Comment (kalexand-rh) [reply modules/machineset-yaml-gcp.adoc]: @sunzhaohua2, you know what "w-a" means here?

Comment (sunzhaohua2) [reply modules/machineset-yaml-gcp.adoc]: I think `w` means worker, `a` means zone `us-central1-a`

Comment (kalexand-rh) [reply modules/machineset-yaml-gcp.adoc]: Do you know how this role abbreviation is obtained? (@ingvagabund, do you know?)

**Comment by vikram-redhat** [comment]
@jhou1 has confirmed this has passed QE.

**Pull request merged**